tests/output/test_printing.py

```diff
@@ -2,72 +2,91 @@
 from textwrap import dedent
 from typing import NamedTuple
-from unittest import mock
+from unittest.mock import Mock, patch
 
 from click import Context, Command
-from snowcli.output.printing import print_db_cursor
+from snowflake.connector.cursor import SnowflakeCursor
+
+from snowcli.output.formats import OutputFormat
+from snowcli.output.printing import print_output, OutputData, print_db_cursor
 
 
 class MockResultMetadata(NamedTuple):
     name: str
 
 
-class _MockCursor:
-    def __init__(self):
-        self._rows = [
-            ("string", 42, ["array"], {"k": "object"}, datetime(2022, 3, 21)),
-            ("string", 43, ["array"], {"k": "object"}, datetime(2022, 3, 21)),
-        ]
-        self._columns = [
-            MockResultMetadata("string"),
-            MockResultMetadata("number"),
-            MockResultMetadata("array"),
-            MockResultMetadata("object"),
-            MockResultMetadata("date"),
-        ]
+def test_print_db_cursor_table(capsys):
+    print_output(OutputData(OutputFormat.TABLE, cursor=_create_mock_cursor()))
 
-    def fetchall(self):
-        yield from self._rows
+    assert _get_output(capsys) == dedent(
+        """\
+    +---------------------------------------------------------------------+
+    | string | number | array     | object          | date                |
+    |--------+--------+-----------+-----------------+---------------------|
+    | string | 42     | ['array'] | {'k': 'object'} | 2022-03-21 00:00:00 |
+    | string | 43     | ['array'] | {'k': 'object'} | 2022-03-21 00:00:00 |
+    +---------------------------------------------------------------------+
+    """
+    )
 
-    @property
-    def description(self):
-        yield from self._columns
 
+def test_print_multi_cursors_table(capsys):
+    mock_cursor = _create_mock_cursor()
 
-@mock.patch("snowcli.output.printing.click.get_current_context")
-def test_print_db_cursor_table(context_mock, capsys):
-    context = Context(Command("foo"))
-    context.params = {"output_format": "TABLE"}
-    context_mock.return_value = context
+    output_data = OutputData(OutputFormat.TABLE)
+    output_data.add_cursor(mock_cursor)
+    output_data.add_cursor(mock_cursor)
 
-    cur = _MockCursor()
-    print_db_cursor(cur)
+    print_output(output_data)
 
-    captured = capsys.readouterr()
-    assert captured.out == dedent(
+    assert _get_output(capsys) == dedent(
         """\
     +---------------------------------------------------------------------+
     | string | number | array     | object          | date                |
     |--------+--------+-----------+-----------------+---------------------|
     | string | 42     | ['array'] | {'k': 'object'} | 2022-03-21 00:00:00 |
     | string | 43     | ['array'] | {'k': 'object'} | 2022-03-21 00:00:00 |
     +---------------------------------------------------------------------+
+    +---------------------------------------------------------------------+
```

Comment:
Should we print query between tables? It will help users to navigate the output. WDYT?

Reply:
I like the idea, but it should be new task. I will create